text/0000-mixed-shadow-mode.md

```diff
@@ -0,0 +1,196 @@
+
+### Lifecycle timing
+
+Due to the way that slotting is implemented in `@lwc/synthetic-shadow`, there is an observable
+difference in the timing of lifecycle hooks for slotted elements.
+
+As mentioned previously, in synthetic mode, slotted elements that are never assigned to a slot are
+not rendered. This means that their lifecycle hooks are never invoked.
+
+Another difference is that for synthetic mode, lifecycle hooks are invoked in the order of
+appearance after they are assigned, whereas in native mode, they are invoked in the order of
+appearance in the template.
+
+### this.shadowRoot vs this.template
+
+`this.shadowRoot` and `this.template` reference the component's shadow root in native Shadow DOM
+while `this.template` references the component's shadow root in synthetic Shadow DOM. There are no
+plans to enable `this.shadowRoot` for existing components that only support synthetic Shadow DOM as
+the current LWC-ism is to use `this.template`.
+
+### this.shadowRoot instanceof ShadowRoot
+
+Components that favor native Shadow DOM will observe that `this.shadowRoot instanceof ShadowRoot`
+evaluates to `false` in mixed mode. This is due to the fact that the synthetic shadow polyfill
+patches and replaces `ShadowRoot`. This will be remediated by customizing the `instanceof` behavior
+using `Symbol.hasInstance`.
+
+### Listening for non-composed events above the root LWC node
+
+There currently exists logic that allows listeners to handle non-composed events outside of the root
+LWC node if the event originates from a non-LWC component in the subtree. This behavior exists for
+legacy reasons and cannot be preserved for components that prefer native Shadow DOM.
+
+### Accessibility
+
+The most common accessibility issue in LWC is related to id-referencing across shadow boundaries.
+The current workaround for this is to bypass the LWC engine's id-mangling by setting attributes
+dynamically, but such a workaround would not work in native Shadow DOM.
+
+### Instrumentation
+
+Applications that rely on instrumentation libraries that don't yet support Shadow DOM are currently
+able to obtain references to descriptors like `addEventListener()` before they are patched and use
+them to override `@lwc/synthetic-shadow` polyfills where needed. Such workarounds would not work for
+events originating from components that prefer native Shadow DOM.
+
+### Styling
+
+Applications currently rely on global stylesheets to apply deeply throughout the DOM. With native
+Shadow DOM, this would not be possible. All of a component's styles would have to be generated and
+added to the component bundle.
+
+## Drawbacks
+
+A potential (non-verified) drawback is that existing components may experience a slight performance
+degradation due to the fact that mixed mode requires logic-forking whenever a Shadow DOM API is
+invoked.
+
+## Alternatives
+
+Other alternatives have not been considered but contributions to this proposal are welcome.
+
+## Adoption strategy

```

Comment:
To make the adoption of mixed-mode successful we will need to provide developers a way to test their components in both modes. 

As of today synthetic shadow is [always loaded](https://github.com/salesforce/lwc-test/blob/master/packages/%40lwc/jest-preset/src/setup.js#L1) in `@lwc/jest-preset`. It would be great if we were to have a way to conditionally load synthetic shadow (https://github.com/salesforce/lwc-test/issues/93). We would then recommend running the same tests with and without synthetic shadow to make sure the component works as expected in both modes.